docs/rsdl/rapid-pro-rsdl-intro.md

```diff
@@ -3,80 +3,238 @@ id: rsdl-intro
+| DELETE http://rapid-pro.org/competitors/cgswl                                         | delete the competitor with the stock symbol cgswl                                               |
+|
+
+### Defining an Enum
+
+Enumerations allow us to define a string-valued property with a fixed set of values.
+
+Let's say that we wanted to define an employmentType enumeration, with possible values "salaried" and "hourly". We could do so as follows:
+
+~~~json
+enum employmentType { salaried hourly }
+~~~
 
-## Defining enums in RSDL
+Now we could use that employmentType enum in our employees example:
 
-## Defining types in RSDL
+~~~rsdl
+type employee
+{
+    @key id: integer
+    firstName: string
+    lastName: string
+    title: string
+    employeeType: employmentType
+}
+~~~
 
-### filtering collections of objects
+### Defining Structural Properties
 
-### Expanding properties
+Our employee has first name and last name properties.  We could define a "fullName" type to group those properties together:
 
-## Defining top level collections of objects
+~~~rsdl
+type fullName
+{
+    firstName: string
+    lastName: string
+}
+~~~
+
+and then use that type in our employee:
 
-## Defining top level single objects
+~~~rsdl
+type employee
+{
+    @key id: integer
+    name: fullName
+    title: string
+}
+~~~
 
-## Querying a graph of objects
+### Defining methods
+RAPID supports functions and actions.
 
-### Types with keys vs types without
+A function takes zero or more input parameters, and returns a value.  Functions must not have side-affects.
 
-### Navigation from one object to another
+We can define a "topEmployees" function on our company:
 
-## Changing data
+~~~rsdl
+type company
+{
+    @key stockSymbol: string
+    name: string
+    incorporated: dateTime
+    employees: [employee]
+    topEmployees(num: integer) : [employee]
+}
+~~~
 
-### create
+topEmployees takes a single integer parameter "num" and returns a collection of employees.
 
-### update
+Functions are invoked using a GET request. Function parameters are passed in the URL.
 
-### delete
+| Request                                                                                              | Comment                                                          |
+| :--------------------------------------------------------------------------------------------------- | :--------------------------------------------------------------- |
+| GET [http://rapid-pro.org/company/topEmployees?num=10](https://jetsons.azurewebsites.net/company/Jetsons.Models.topEmployees(num=10))  | get the company's top 10 employees                             |
+|
+An action takes zero or more input parameters and may or may not return a value.  Actions may have side-affects.
+
+We can define a "youreFired" action on our company that takes a string parameter "reason":
```

Comment:
```suggestion
We can define a "yourFired" action on our company that takes a string parameter "reason":
```

Reply:
Fixed -- actually, it should be "youreFired" in each instance.